phoenix-core/src/main/java/org/apache/phoenix/jdbc/PhoenixEmbeddedDriver.java

```diff
@@ -225,16 +233,44 @@ private static boolean isMultiPortUrl(String portStr) {
             }
             return false;
         }
-        
+
         public static ConnectionInfo create(String url) throws SQLException {
             url = url == null ? "" : url;
             if (url.isEmpty() || url.equalsIgnoreCase("jdbc:phoenix:")
                     || url.equalsIgnoreCase("jdbc:phoenix")) {
                 return defaultConnectionInfo(url);
             }
+
+            // ex: jdbc:phoenix+hrpc:hostname1....
             url = url.startsWith(PhoenixRuntime.JDBC_PROTOCOL)
                     ? url.substring(PhoenixRuntime.JDBC_PROTOCOL.length())
                     : PhoenixRuntime.JDBC_PROTOCOL_SEPARATOR + url;
+
+            PhoenixRuntime.HBaseBootstrap bootstrapAsEnum = null;
+
+            // ex: +hrpc:hostname1....
+            if (url.startsWith(String.valueOf(PhoenixRuntime.JDBC_PROTOCOL_CONNECTOR_PREFIX))) {
+                final String firstToken = url.split(String.valueOf(PhoenixRuntime.JDBC_PROTOCOL_SEPARATOR))[0];
```

Comment:
Suggest not using `String.split(String)` as this ends up being a regex operation.

Please use `int offset = url.indexOf(PhoenixRuntime.JDBC_PROTOCOL_SEPARATOR)` and `url.subString(0, offset)` to extract the first piece (or use a tokenizer like elsewhere in this file).